specification/cognitiveservices/OpenAI.Inference/models/completions.create.tsp

```diff
@@ -17,7 +17,7 @@ during training, so if a prompt is not specified the model will generate as if
 from the beginning of a new document. Maximum allowed size of string list is
 2048.
 """)
-    prompt?: string[];
+    prompt?: string | string[];
```

Comment:
This will break some languages we should keep it as `string[]` until we have finalized our client.tsp definitions on these types of overloads.

Given that string is inside the domain of string[] this shouldn't be an issue and if we want a simple method which only takes a string we are doing that with custom code in dotnet today.

Reply:
@m-nash, could you elaborate on how things will break? The .NET SDK already has an overload that expects a string as input ([here](https://github.com/Azure/azure-sdk-for-net/blob/c552ef6c7e92f30f3e92efc51312e6f53f65ee3a/sdk/openai/Azure.AI.OpenAI/src/Custom/OpenAIClient.cs#L178)), wouldn't this change generate a similar overload so the custom one is no longer needed?

Reply:
We need to open a high-priority bug on the emitters that cannot deal with this. It is very much a fairly urgent goal to make the specification match the actual service.

Reply:
If that means generate as internal-only with a System.Object typed parameter, and we wrap the methods for now, that should be doable, no?

Reply:
@johanste I don't think leaving it as is blocks anything we are simply dealing with this as custom code for now.  Can you help me understand what is broken if we leave it as is?

If it is mission critical to change this we could try it with `string[] | string` as I think right now dotnet simply grabs the first member of the union and ignores the rest so that we don't crash.  Definitely changing it to `string | string[]` will break.

@srnagar How will java deal with adding this back?

Reply:
> @m-nash, could you elaborate on how things will break? The .NET SDK already has an overload that expects a string as input ([here](https://github.com/Azure/azure-sdk-for-net/blob/c552ef6c7e92f30f3e92efc51312e6f53f65ee3a/sdk/openai/Azure.AI.OpenAI/src/Custom/OpenAIClient.cs#L178)), wouldn't this change generate a similar overload so the custom one is no longer needed?

Yes this is the custom code that I described above.  We removed the union in the spec and dealt with it in custom code for now.

Reply:
I switched the order.